srml/example/src/lib.rs

```diff
@@ -258,13 +258,52 @@ use support::{dispatch::Result, decl_module, decl_storage, decl_event};
 use system::{ensure_signed, ensure_root};
 use codec::{Encode, Decode};
 use sr_primitives::{
-	traits::{SignedExtension, Bounded}, weights::{SimpleDispatchInfo, DispatchInfo},
+	traits::{SignedExtension, Bounded, SimpleArithmetic, SaturatedConversion},
+	weights::{SimpleDispatchInfo, DispatchInfo, DispatchClass, ClassifyDispatch, WeighData, Weight},
 	transaction_validity::{
 		ValidTransaction, TransactionValidityError, InvalidTransaction, TransactionValidity,
 	},
 };
 
-/// Our module's configuration trait. All our types and consts go in here. If the
+// A custom weight calculator tailored for the dispatch call `set_dummy()`. This actually examines
+// the arguments and makes a decision based upon them.
```

Comment:
You assume here the reader knows what transaction weight is. I think you can elaborate on "decision" either talking about the max weight of a block or that weights trigger associated fees, or whatever is actually accurate :)

Reply:
So I would keep that for a developer hub page about weights. For the sake of your concern, there is another full paragraph in the same page about _what transaction weight is_. I will keep this file simple and short as it is not really intended to be a root documentation. 

What I really want to achieve here is just to have one working example of a common scenario, as Joshy faced.